### DISCUSSION TITLE
please help

### DISCUSSION THREAD

**Discussion by despotgit** [dwatteau/scummtr#61] [Q&A]
I am using the commands for export and import of the text, but when I add the game in ScummVM, should I select a language? I am translating the english dialogues to serbian (regular latin, nothing fishy). My problem is that the text is no longer displayed when I load the game. CTRL+T doesn't give the option at all, no options actually! No voice, no text no voice+text options. This is the command that I use to import

../scummtr -g monkey2 -cwh -A aov -i

and it seems to be working fine (there is a little bit of time, and then the command prompt returns, as if it is finished). Should the scummtr.txt be in a specific encoding? I tried not changing it (which is by default UTF), but I also tried setting the encoding to lossless-ASCII, and the problem is still the same, the texts are not there any more.

**Comment by dwatteau** [comment]
Hello @despotgit!

Thank you for opening this first discussion in this space :)

I'll have a look at your problem when I can find a bit of free time for it, in the following days.

Meanwhile, can you tell me which text editor and system you are using?

Per [our FAQ](https://github.com/dwatteau/scummtr/blob/main/FAQ.md) I'd suggest using something like Notepad++ if you're on Windows, for example. scummtr is currently very sensitive to the automatic changes made by some text editors (newlines, whitespace, encoding…) so it's recommended to use a text editor which is known to "respect" the original format.

As for the encoding, for West-European languages, scummtr outputs to Windows-1252/ISO-8859-1 (sometimes also known as "ANSI" in the Windows world). But if you run scummtr with the `-c` flag, you should have a header in the first lines of your translation file, helping your text editor recognize the encoding we want.

So, along with your text editor / OS, can you also show the first few lines of your translations files?

If you don't see any encoding header in your translation file, it may mean you're running the old 2003 release of scummtr. In this case, please update to [the latest release](https://github.com/dwatteau/scummtr/releases), which has several improvements.

(EDIT : miss-click, didn't meant to close the discussion)

Also, which edition of Monkey Island 2 are you trying to translate? The original DOS release? The Talkie one? (a.k.a. Ultimate Talkie Edition?)

The more context you give, the easier it is to give you a quick reply helping you solve the issue. Thanks!

**Reply by despotgit** [reply]
Hello my good man!!!!! Meanwhile I managed to do it!!!! omg :) such hapiness that you have brought upon me. Hahahaha yea you have. I think the error was just that the lengths of some of my translations were different than the original ones, causing the problems. Speaking of which, how could I overcome this issue (sometimes I nee longer text for the serbian translation)? I am using the Ultimate Talkie MI2, yes :)

**Reply by dwatteau** [reply]
Cool @despotgit 

What do you mean by "length"?

* If you mean "Number of lines in the translation files": yes, you can't add or remove lines in the text translation file. You need to keep exactly the same number of lines.
    * (Actually, there _are_ ways of adding or removing lines in the script, but it's quite technical. If your intent is, for example, to add subtitles to the "Part I", "Part II" screens of the game, I can help you do that, later on.)
* If you mean "number of characters in a line", then there shouldn't be immediate issues. Your text may appear truncated (I'll explain below how to deal with that), but it shouldn't be completely missing.

Can you show a screenshot of your problem(s) in the game? And send your raw, complete `.txt` translation file in Serbian?

**Reply by dwatteau** [reply]
Ah, you're running the Ultimate Talkie edition, and you have no sound anymore when you import a new translation?

Yeah, if that's the release you're trying to translate, there might be reasons for Ctrl-T not working anymore. That's because you're probably modifying it first, and adding it to ScummVM next. You should do the opposite, so that it gets properly detected as the Talkie release, while you're working on it.

Here's how I recommend working:

* keep a backup copy of the original, English release of the Ultimate Talkie Edition. _Never_ modify it. Just keep it around, unmodified, it's your reference point and your safety belt.
* make a copy of this original directory. Call this new copy "MONKEY2_ULTIMATE_SERBIAN" or something like that. Add it like this to ScummVM, _before_ making any change to it. ScummVM should detect it properly.
* Then, start importing your translation file in this copy. This way, ScummVM detected that's it's a Talkie release, and the Ctrl-T setting should continue working, even after making changes to it.
* Once you're fully done with your Serbian translation, it may get added to the ScummVM detection list, as long as you don't redistribute your fan translation as a _full game_, only as a patch. 
* always make regular backups while you work on your translation. Really do. You have a high risk of losing hours of work if you misuse scummtr at some point, and if the game gets corrupted in some way. Just Ctrl-C/Ctrl-V the whole directory containing your work, regularly, somewhere. If anything goes wrong, you'll be happier to loose a couple of hours of works, instead of weeks of work, trust me ;)

**Reply by despotgit** [reply]
Hi. Regarding the saving of the work issue....I am actually tracking it in git :) every change i make in scummtr.txt I am committing to git. After having tested it of course.

**Reply by dwatteau** [reply]
As for dealing with "long sentences", see this part of the FAQ:

https://github.com/dwatteau/scummtr/blob/main/FAQ.md#when-a-sentence-is-too-long-the-game-doesnt-display-it-properly

Basically, you need to insert `\255\001` when you want your text to return to a new line, before it overflows on the screen.

(In general, I _highly_ recommend reading the whole FAQ. It's a bit long, but it should help you with various common problems.)

Anyway, in Monkey Island 2, you may run into issues if your "long sentences" are dialogue options. You'll see glitches when you hover the cursor over such a long sentence, if the original sentence didn't use the same number of lines.

There are ways of fixing this, too. But again, it's a bit technical. Before I help you with this, I need your answers on the questions above, though :)

**Reply by despotgit** [reply]
Hi dwatteau, so, if the length of the line is different, the ctrl-T doesn't work any more, and there is no dialog texts any more. None. But, I take care and I add always spaces if my serbian translation is shorter. However, my problem is when I wanna put serbian translatiaon that is actually LONGER than the original english. ANd I sometimes need to do it, so...problems arise that I have to cut some letters out. For example in word "Push" for the verbs in the verb toolbox, I need to put "Gurni" and I cannot, so I put a little but ugly cut-off word "Gurn"......etc.

**Reply by dwatteau** [reply]
@despotgit I can help you with that, yes, but as asked earlier , please provide examples of:

* lines being too long (e.g. an extract of your .txt file)
* a screenshot showing your problem in-game

so that I can reproduce it locally and tell you how you can deal with this.

For the verbs in particular, yes, its script needs to be updated, I can help you with that if you give me the details above. (You could also reuse what the "SCUMMPL" fan-translation of Monkey Island 2 did for its verbs; they changed the font for the longer words of Polish.)

**Reply by despotgit** [reply]
This is my current scummtr.txt in attachment
[scummtr.txt](https://github.com/user-attachments/files/17112318/scummtr.txt)
And I attach the screenshot of shortened verb pallette. E.g. I need also to change the serbian word for "Close" to "Zatvo" instead of the correct "Zatvori".
![Screenshot 2024-09-24 at 11 54 30](https://github.com/user-attachments/assets/3941d2b6-cc90-4db2-8c3e-f191d514cdbe)

**Reply by dwatteau** [reply]
> This is my current scummtr.txt in attachment
> 
> [scummtr.txt](https://github.com/user-attachments/files/17112318/scummtr.txt)
> 
> And I attach the screenshot of shortened verb pallette. E.g. I need also to change the serbian word for "Close" to "Zatvo" instead of the correct "Zatvori".
> 
> ![Screenshot 2024-09-24 at 11 54 30](https://github.com/user-attachments/assets/3941d2b6-cc90-4db2-8c3e-f191d514cdbe)
> 
> 

Perfect! Thanks, this is exactly what I need to properly investigate your problem

I'm at a work seminar this week but I'll try having a look at this as soon as possible 

Hold on ;)

**Reply by dwatteau** [reply]
@despotgit In the mean time, can you give me an example of a line (or place in the game) when having a long line makes Ctrl-T stop working for you?

Thanks!

**Reply by despotgit** [reply]
@dwatteau  pretty much anywhere if I add an extra character, the ctrl-T doesn't work any more, and no text is displayed.
I add a letter, do my script:
scummtr -g monkey2 -i -f scummtr.txt
and ctrl-T stops working.
One example of a line:
\255\010\201\147\255\010\166\000\255\010\010\000\255\010\000\000A 'de  si  ti krenuo, Lepice-Pantalonice? m   
(Here I added an "m" character)
